src/components/LicensesFieldArray/AmendmentsFieldArray.js

```diff
@@ -35,40 +35,35 @@ class AmendmentsFieldArray extends React.Component {
     name: PropTypes.string.isRequired,
   };
 
-  warnStatusMismatch = ({ values }) => {
-    const { form, name } = this.props;
+  constructor(props) {
+    super(props);
+    this.state = { statusWarnings: [] };
```

Comment:
So, per the comments in the review body, this isn't necessary at all, but if it _was_, you could write this a lot cleaner via something like the following:
```
const { items = [] } = props; // guarantees items exists and is an empty array so we can perform array methods on it

this.state = {
  warnings = items.map(() => ({ warningId: '', warningValuesStatus: null }))
}
```
There aren't a tonne of situations where a for-loop is actually the preferred way of writing a loop in JS. 

Also, falsey values can be a really tricky situation in JS, so avoid `null` except where you're _definitely_ sure you specifically want a `null` as opposed to the much more common `undefined`. Empty strings are useful for when you want a falsey value to call string methods on a variable without checking if it's a string (vs undefined) or if you're using it in a template literal. Ie, ``` `${foo}` ``` => `"undefined"` if foo is undefined, rather than just an empty string.

Reply:
Duly noted, as above this has now been removed anyway.